lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -103,47 +105,21 @@ def identify(self, requirement_or_candidate):
         """
         return requirement_or_candidate.canonical_package_id
 
-    def get_preference(self, *args, **kwargs):
-        # type: (t.Any, t.Any) -> t.Union[float, int]
+    def get_preference(
+        self,
+        identifier: str,
+        resolutions: t.Mapping[str, Candidate],
```

Comment:
Abstract container types like `Mapping` and `Iterator` in `collections.abc` have supported subscripting since Python 3.9. We should be using those in controller code instead of the deprecated ones in the `typing` module. In ansible-core 2.20 we'll be able to use them for target code as well.